src/ServicePulse.Host/vue/src/components/failedmessages/EditRetryDialog.vue

```diff
@@ -1,26 +1,36 @@
-<script setup>
+<script setup lang="ts">
 import { computed, onMounted, ref, watch } from "vue";
 import { useRetryEditedMessage } from "../../composables/serviceFailedMessage";
 import MessageHeader from "./EditMessageHeader.vue";
+import { FailedMessageViewWithExtendedUIProperties } from "@/resources/FailedMessageView";
+import { EditAndRetryConfig } from "@/resources/Configuration";
+import type Header from "@/resources/Header";
+
+interface HeaderWithEditing extends Header {
+  isLocked: boolean;
+  isSensitive: boolean;
+  isMarkedAsRemoved: boolean;
+  isChanged: boolean;
+}
 
 const emit = defineEmits(["cancel", "retried"]);
```

Comment:
You can type emits, see https://vuejs.org/guide/typescript/composition-api.html#typing-component-emits